.circleci/config.yml

```diff
@@ -1,65 +1,142 @@
+    docker:
+      - image: quay.io/deis/lightweight-docker-go:v0.2.0
+    environment:
+      SKIP_DOCKER: true
+      CGO_ENABLED: 0
+    working_directory: /go/src/github.com/Azure/open-service-broker-azure
     steps:
       - checkout
+      # This is a workaround for a problem I (krancour) cannot pin down. The
+      # linter works flawlessly in all environments (containers included)
+      # EXCEPT a container in Circle. I've remoted into running containers in
+      # Circle and have attempted, at great length, to diagnose the issue, but
+      # am unable to do so. What I did discover, however, was that by
+      # re-installing gometalinter (and the linters it, in turn, uses), I can
+      # make the problem go away. ODDLY, when comparing the binaries that were
+      # included in the image to the NEW binaries, diff says they are identical.
+      # I'm at a complete loss for why this happens. I can live with this
+      # workaround for now because I anticipate our days using Circle for CI
+      # are numbered anyhow.
+      - run:
+          name: Re-Install Gometalinter
+          command: |
+            rm /go/bin/*
+            cd /go/src/github.com/alecthomas/gometalinter
+            go clean -i
+            go install
+            gometalinter --install
       - run:
           name: Run Lint
           command: make lint
   verify-vendored-code:
-    machine: true
+    docker:
+      - image: quay.io/deis/lightweight-docker-go:v0.2.0
+    environment:
+      SKIP_DOCKER: true
+      CGO_ENABLED: 0
+    working_directory: /go/src/github.com/Azure/open-service-broker-azure
     steps:
       - checkout
       - run:
           name: Verify Vendored Code
           command: make verify-vendored-code
   test-unit:
-    machine: true
+    docker:
+      - image: quay.io/deis/lightweight-docker-go:v0.2.0
+      - image: redis:3.2.4
+    environment:
+      SKIP_DOCKER: true
+      STORAGE_REDIS_HOST: localhost
+      ASYNC_REDIS_HOST: localhost
+      CGO_ENABLED: 0
+    working_directory: /go/src/github.com/Azure/open-service-broker-azure
     steps:
       - checkout
       - run:
           name: Run Unit Tests
           command: make test-unit
   test-api-compliance:
-    machine: true
+    docker:
+      - image: quay.io/deis/lightweight-docker-go:v0.2.0
+    environment:
+      SKIP_DOCKER: true
+      CGO_ENABLED: 0
+      TESTS_PATH: /opt/osb-checker/2.13/tests
+    working_directory: /go/src/github.com/Azure/open-service-broker-azure
     steps:
       - checkout
+      # This is a workaround. See comments in the script.
+      - run:
+          name: Copy Modified Test Scripts
+          command: |
+            mkdir -p /app
+            cp tests/api-compliance/scripts/* /app/
       - run:
           name: Run API Compliance Tests
-          command: make test-api-compliance
+          command: |
+            go run cmd/compliance-test-broker/compliance-test-broker.go &
+            make test-api-compliance
```

Comment:
These need to be in the same step. I tried to break them up, but even with `nohup`, I couldn't get the broker to keep running between steps.